cmd/otelcol/flags_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright  Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFlagParseFailure(t *testing.T) {
+	oldArgs := os.Args
+
+	// Parsing should stop once an unspecified flag is found
+	os.Args = []string{"otelcol", "--invalid-flag", "100", "--version", "true"}
+	inputFlags, _ := parseFlags(os.Args[1:])
```

Comment:
This can introduce some maintenance overhead but it's helpful to capture what we expect to be the case for possible errors. Similarly w/ `assert.NoError(t, err)`:

```suggestion
	inputFlags, err := parseFlags(os.Args[1:])
	assert.EqualError(t, err, "expected content")
```

Reply:
I've updated.